docs/tinymist/frontend/neovim.typ

```diff
@@ -22,121 +26,81 @@ See #link("https://github.com/Myriad-Dreamin/tinymist#features")[Tinymist Featur
-You can edit the `coc-settings.json` by executing `:CocConfig`:
-
-```json
-{
-  "languageserver": {
-    "tinymist": {
-      "command": "tinymist",
-      "filetypes": ["typst"],
-      "settings": { ... }
+- Or manually:
+
+ #include "common-finding-executable.typ"
+
+== Configuration
+- With `lspconfig` 
+    ```lua
+    require("lspconfig")["tinymist"].setup {
+        settings = {
+            tinymist = {
+                settings = {
+                    formatterMode = "typstyle",
+                    exportPdf = "onType",
+                    semanticTokens = "disable"
+                    -- ...
+                },
+            },
+        },
     }
-  }
-}
-```
-
-== Tips
-<tips>
-=== Working with Multiple-File Projects
-<working-with-multiple-file-projects>
-The solution is a bit internal, which should get further improvement, but you can pin a main file by command.
-
-```lua
--- pin the main file
-vim.lsp.buf.execute_command({ command = 'tinymist.pinMain', arguments = { vim.api.nvim_buf_get_name(0) } })
--- unpin the main file
-vim.lsp.buf.execute_command({ command = 'tinymist.pinMain', arguments = { nil } })
-```
+    ```
+
+- Or with `Coc.nvim`
+
+    ```json
+    {
+      "languageserver": {
+        "tinymist": {
+          "command": "tinymist",
+          "filetypes": ["typst"],
+          "settings": { ... }
+        }
+      }
+    }
+    ```
+- Or finally with the builtin lsp protocol
+    
+    ```lua
+    vim.lsp.config['tinymist'] = {
+        cmd = {'tinymist'}, 
+        filetypes = {'typst'}
+        settings = {
+            -- ...
+        }
+    }
+    ```
+    For a full list of availible settings see #link("https://github.com/Myriad-Dreamin/tinymist/tree/main/editors/neovim/Configuration.md")[Tinymist Server Configuration]. 
 
-There is also a plan to support multiple-file projects with a workspace configuration, but I don’t know whether it is Neovim’s way, so it needs further discussion.
+== Formatting
+    Install either `typststyle` or `typstfmt` either via source or via `Mason`. Then specify the formatter in your config
 
-#note-box[
-  `tinymist.pinMain` is a stateful command, and tinymist doesn't remember it between sessions (closing and opening the editor).
-]
+    ```lua 
+    formatterMode = "typstyle"
+    ```
+== Preview
+    There is work in progress to integrate #link("https://github.com/chomosuke/typst-preview.nvim")[typst-preview.nvim] directly into `tinymist`. In the meantime you can find can find installation and configuration instructions on that repo. This should be a shortterm solution. 
 
 == Troubleshooting
 <troubleshooting>
-=== tinymist does not start on creating/opening files
-<tinymist-does-not-start-on-creatingopening-files>
-First, please check that tinymist starts when manually setting the filetype.
 
-```
-:set filetype=typst
-```
+Generally you can find in depth information via the `:mes` command. Tinymist also creates a debug log that is usually at `~/.local/state/nvim/lsp.log`  
```

Comment:
What is a `:mes`, is it a builtin command? I remember there is also a builtin `:checkhealth` command. Which one is better?

Reply:
:mes is usually more direct but you are right that both are useful